pinot-core/src/main/java/org/apache/pinot/core/query/pruner/DataSchemaSegmentPruner.java

```diff
@@ -35,7 +38,9 @@ public void init(PinotConfiguration config) {
 
   @Override
   public boolean prune(IndexSegment segment, QueryContext query) {
-    return !segment.getColumnNames().containsAll(query.getColumns());
+    Set<String> columns =
+        query.getColumns().stream().map(col -> StringUtils.split(col, "$", 2)[0]).collect(Collectors.toSet());
```

Comment:
This is already a common bottleneck when there are large numbers of segments, I can show you some profiles, this will make it worse,

Reply:
Agreed, I will add those columns to the schema